File: 8kyu/8kyu 1/hello.js
function hello(name = "") {
   if (name == "") {
      return `Hello, World!`;
   } else {
      const lower = name.toLowerCase().substr(1);
      const upper = name[0].toUpperCase();
      return `Hello, ${upper + lower}!`;
   }
}


function hello(name = "") {
   if (name == "") {
      return `Hello, World!`;
   } else {
      return `Hello, ${name.charAt(0).toUpperCase() + name.toLowerCase().substr(1)}!`
   }
}


function hello(name) {
   if (name) {
      return "Hello, " + name.substring(0, 1).toUpperCase() + name.substring(1).toLowerCase() + '!';
   } else {
      return "Hello, World!";
   }
}


function hello(name) {
   return `Hello, ${name ? name[0].toUpperCase() + name.slice(1).toLowerCase() : "World"}!`
}


console.log(hello('johN'), 'Hello, John!')
console.log(hello('alice'), 'Hello, Alice!')
console.log(hello(), 'Hello, World!')
console.log(hello(''), 'Hello, World!')

File: Python/8py 2/hello.py
"""
|--------------------------------------------------------------------------
| Hello, Name or World!
|--------------------------------------------------------------------------
|
"""

def hello(name=None):
   if name is None or name == '':
      return 'Hello, World!'
   else:
      return f'Hello, {name.capitalize()}!'


hello = lambda name="": "Hello, {}!".format("World" if not name else name.capitalize())


def hello(name=''):
   return f"Hello, {name.title() or 'World'}!"


def hello( name = ''):
   return 'Hello, World!' if len(name) == 0 else 'Hello, ' + name[0].upper() + name[1:].lower() + '!'


def hello(name=''):
   return "Hello, {}!".format(name.title() if name else 'World')


print(hello('johN'), 'Hello, John!')
print(hello('alice'), 'Hello, Alice!')
print(hello(), 'Hello, World!')
print(hello(''), 'Hello, World!')